### PR TITLE
Add getter/setter methods for Domain Registry contract

### PR DESCRIPTION
**Pull request by stephankmin**

### CODE REVIEW

File: src/seaport.ts [hunk 1]
```diff
@@ -932,4 +943,27 @@ export class Seaport {
...
+    const signer = this._getSigner(accountAddress);
+
+    return getTransactionMethods(
+      this.domainRegistry.connect(signer),
+      "setDomain",
+      [domain]
+    );
+  }
+
+  public getDomain(tag: string, index: number): Promise<string> {
```
Comment (0age) [inline src/seaport.ts:962]: Should include a getTotalDomains here

File: src/seaport.ts [hunk 2]
```diff
@@ -932,4 +943,27 @@ export class Seaport {
...
+    return this.domainRegistry.getDomain(tag, index);
+  }
+
+  public async getDomains(tag: string): Promise<string[]> {
+    return this.domainRegistry.getDomains(tag);
```
Comment (0age) [inline src/seaport.ts:967]: For this one, I'd say try/catch it and if it fails then fall back to getTotalDomains + map(getDomain)

**Review by 0age — CHANGES_REQUESTED** [review]
Small change to the getter proposed

**Review by 0age — APPROVED** [review]

File: src/seaport.ts [hunk 3]
```diff
@@ -932,4 +944,54 @@ export class Seaport {
       domain
     );
   }
+
+  public setDomain(
+    domain: string,
+    accountAddress: string
```
Comment (sunwrobert) [inline src/seaport.ts:950]: this can be optional?

File: src/seaport.ts [hunk 4]
```diff
@@ -932,4 +944,54 @@ export class Seaport {
...
+  public async getDomains(
+    tag: string,
+    shouldThrow?: boolean
+  ): Promise<string[]> {
+    try {
+      if (shouldThrow) {
+        throw Error;
+      }
+
+      this.domainRegistry.getDomains(tag);
```
Comment (sunwrobert) [inline src/seaport.ts:980]: why don't we just return here? so we don't need to return below.

**Review by sunwrobert — APPROVED** [review]

**Pull request merged**